### PR TITLE
Add Feature Documentation

### LINKED ISSUE #153: Write Docs on How To Add A Feature/Functionality
Steps are roughly:

- [ ] Find event that corresponds to the desired feature
- [ ] Decide if there are any pre-requisites to the feature being triggerwd (think: focused window only, open tab only, etc.)
- [ ] See if the prerequisite is already defined (we have some for only focused window events for example)
- [ ] Decide what, if any, state is required for your feature (caret position, last focused item, etc.)
- [ ] Check to see if we already have your state info as a type (if not, make a newtype and implement FromAsyncState for it)
- [ ] Implement a new function that takes `PreRequisiteType<EventType>`, and `StateType`. That function can retuen a list of `Command`s that Odilia will act on.

### PR DESCRIPTION
**Pull request by TTWNO**
Fix #153

### CODE REVIEW

**Comment by albertotirla** [comment]
alright, it looks good from my perspective, however there's one more thing that could be done if you want.
Optionally, you could add more details about what it means to make a type and implement FromAsyncState for it, basically point the contributors to a minimal example where this is implemented, via a link they can click on to get to the files, perhaps the handler for focused events is simple enough?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,6 +10,27 @@ These can be added to your pre-commit hooks to automate the checks. Beyond these
...
 
+## How To Add A Feature
+
+Odilia has a unique architecture which embeds alot of code into types.
+Here is what you need to know if you want to add a new feature:
+
+- [ ] Find the AT-SPI event that corresponds to the desired feature
+    - Take a look at the docs for `common/events/` and try to find the right event.
+    - For example, to create a feature that reads out changed text in an aria-live region, you would want the event `TextChangedEvent`.
+- [ ] Decide if there are any pre-requisites to the feature being triggerwd (think: focused window only, open tab only, etc.)
```
Comment (ferdnyc) [inline CONTRIBUTING.md:21]:
```suggestion
- [ ] Decide if there are any prerequisites to the feature being triggered (think: focused window only, open tab only, etc.)
```

Spelling, consistency

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -10,6 +10,27 @@ These can be added to your pre-commit hooks to automate the checks. Beyond these
...
+- [ ] See if the prerequisite is already defined
+    - Check out the `odilia/src/tower/cache_event.rs` file for types that implement the `Predicate` trait (defined in the `refinement` create).
+- [ ] Decide what, if any, state is required for your feature (caret position, last focused item, etc.)
+    - For example: to know if the current window is focused, the current window must be stored in the state of the screen reader.
+- [ ] Check to see if we already have your state info as a type; you can check this at `odilia/src/state.rs`.
+    - If not, make a [newtype](https://doc.rust-lang.org/book/ch19-04-advanced-types.html#using-the-newtype-pattern-for-type-safety-and-abstraction), add it to the `State` struct, then implement the `FromAsyncState` trait for the type (so it can be extracted by the `FromAsyncState` implementation).
```
Comment (ferdnyc) [inline CONTRIBUTING.md:27]: I wonder if it wouldn't be better, from a document-source-reading perspective, if this URL were a named reference? You'd change the link here to something like,

```markdown
[newtype][newtype-pat]
```

And waay at the bottom of the file, drop in:
```markdown
[newtype-pat]: https://doc.rust-lang.org/book/ch19-04-advanced-types.html#using-the-newtype-pattern-for-type-safety-and-abstraction
```

Comment (albertotirla) [reply CONTRIBUTING.md:27]: hmm, that's debatable, as most people would read it in rendered form anyway. Definitely could be done though, if @TTWNO thinks it's better that way. If you read the markdown source though, it's definitely better to do it the way you suggest

**Pull request merged**